src/balatro_gym/game/shop.py

```diff
@@ -0,0 +1,61 @@
+import dataclasses
+import random
+from typing import Sequence
+
+from balatro_gym.cards.interfaces import Card
+from balatro_gym.cards.voucher import ALL_VOUCHERS
+from balatro_gym.interfaces import BoosterPack, Voucher
+
+
+@dataclasses.dataclass
+class ShopState:
+    buyable_cards: Sequence[Card]
+    vouchers: Sequence[Voucher]
+    booster_packs: Sequence[BoosterPack]
+
+
+class Shop:
+
+    def __init__(self,
+    num_buyable_slots: int = 2,
+    num_vouchers: int = 1,
+    num_booster_packs: int = 2,
+    reroll_price: int = 5):
+        self.num_buyable_slots = num_buyable_slots
+        self.num_vouchers = num_vouchers
+        self.num_booster_packs = num_booster_packs
+        self.reroll_price = reroll_price
+        self.vouchers: Sequence[Voucher] = []
+        self.bought_vouchers: Sequence[Voucher] = []
+        self.current_state = None
+
+    def increase_num_buyable_slots(self) -> None:
+        self.num_vouchers += 1
+
+    def increase_voucher_limit(self) -> None:
+        self.num_vouchers += 1
+
+    def change_reroll_price(self, price: int) -> None:
```

Comment:
```suggestion
    def set_reroll_price(self, price: int) -> None:
```
or using the `@setter` decorator, to match python naming standards